Algorithms/Common/DayofWeek.cs

```diff
@@ -0,0 +1,157 @@
+﻿using System;
+
+namespace Algorithms.Common
+{
+    public static class DayofWeek
+
+    {
+        /// <summary>
+        /// Determine day of week of the date
+        /// </summary>
+        /// <returns>Return string of day of week of the date</returns>
+        /// <param name="date">Date.</param>
+        public static string GetDayofWeek(DateTime date)
+        {
+
+            int count = 1; //Int to shows Ordinal number of occurrence of day of week
+            DateTime Occurrence = date.AddDays(-7); //Date to check the ordinal number of day of week 
+
+            for (string month = date.ToString("MMMM"); month == Occurrence.ToString("MMMM"); Occurrence = Occurrence.AddDays(-7))
```

Comment:
1. Alot conversion without regional settings.
2. Alot string operations
3. Easy syntax - .ToString("MMMM") -> .Month

Example 1th refactoring
`for (int month = date.Month; month == Occurrence.Month; Occurrence = Occurrence.AddDays(-7))`
Example 2nd refactiong:
`int count = date.Day % 7;`